python/pyspark/resource/profile.pyi

```diff
@@ -22,30 +22,39 @@ from pyspark.resource.requests import (  # noqa: F401
     TaskResourceRequest as TaskResourceRequest,
     TaskResourceRequests as TaskResourceRequests,
 )
-from typing import Any, Optional
+from typing import overload, Dict, Optional, Union
+from py4j.java_gateway import JavaObject  # type: ignore
 
 class ResourceProfile:
+    @overload
     def __init__(
         self,
-        _java_resource_profile: Optional[Any] = ...,
-        _exec_req: Any = ...,
-        _task_req: Any = ...,
+        _java_resource_profile: Optional[JavaObject] = ...,
+    ) -> None: ...
+    @overload
+    def __init__(
+        self,
+        *,
```

Comment:
`ResourceProfile.__init__` seems not a keyword-only. Should we maybe change it back?

Reply:
ohh got it. NVM. the codes assumes it.

Reply:
> ohh got it. NVM. the codes assumes it.

Yeah, it is all about control flow there. But this

```
    @overload
    def __init__(
        self,
        _java_resource_profile: None = ...,
        _exec_req: Dict[str, ExecutorResourceRequest] = ...,
        _task_req: Dict[str, TaskResourceRequest] = ...,
    ) -> None: ...
```

should have the same meaning here.